File: src/main/java/kanti/sl/objects/StateObjectImpl.java
package kanti.sl.objects;

import kanti.sl.arguments.StateArgument;
import kanti.sl.arguments.StateArguments;
import org.jetbrains.annotations.NotNull;

public class StateObjectImpl implements StateObject {

	private final String name;
	private final StateArguments args;

	public StateObjectImpl(@NotNull String name, @NotNull StateArgument... args) {
		this.name = name;
		this.args = StateArguments.create(args);
	}

	@NotNull
	@Override
	public String getName() {
		return name;
	}

	@NotNull
	@Override
	public StateArguments getArguments() {
		return args;
	}

}

File: src/main/kotlin/kanti/sl/objects/StateObjectImpl.kt
package kanti.sl.objects

import kanti.sl.arguments.StateArgument
import kanti.sl.arguments.StateArguments
import kanti.sl.arguments.StateArgumentsImpl

class StateObjectImpl(
	override val name: String,
	args: Iterable<StateArgument>
) : StateObject {

	override val arguments: StateArguments = StateArgumentsImpl(args)

}